v4/export/config.go

```diff
@@ -117,6 +129,53 @@ func (conf *Config) GetDSN(db string) string {
 	return dsn
 }
 
+func (config *Config) ExternalStorage(ctx context.Context) storage.ExternalStorage {
+	config.externalStorageOnce.Do(func() {
+		s, err := config.createExternalStorage(ctx)
+		if err != nil {
+			panic(err)
+		}
+		config.externalStorage = s
+	})
+	return config.externalStorage
+}
+
+func (config *Config) createExternalStorage(ctx context.Context) (storage.ExternalStorage, error) {
+	path := config.OutputDirPath
+	var b *backup.StorageBackend
+	var err error
+	// TODO support other filesystems
+	options := &config.BackendOptions
+	// used for testing
+	endpoint := os.Getenv("DUMPLING_S3_ENDPOINT")
+	if endpoint != "" {
+		options.S3.Endpoint = endpoint
+		options.S3.ForcePathStyle = true
+		options.GCS.Endpoint = endpoint
+	}
+	b, err = storage.ParseBackend(ensureTrailingSlash(ensureStorageClassPrefix(path)), options)
+	if err != nil {
+		return nil, err
+	}
+	// TODO support sendCreds config option?
+	return storage.Create(ctx, b, true)
+}
+
+func ensureStorageClassPrefix(path string) string {
+	if !regexp.MustCompile("^\\w+://").MatchString(path) {
+		return "file:///" + path
```

Comment:
If I use a relative path here, for example, `./export-2020-09-25T16:48:39+08:00`, this function will return `file:///./export-2020-09-25T16:48:39+08:00` which seems incorrect.

Reply:
I will take a look on how to solve this problem.

Reply:
We need to change the relative directory to an abs one. I will fire a PR in BR to solve this problem.

Reply:
https://github.com/pingcap/br/pull/531

Reply:
nice!